eng/common/templates/job/build-from-source.yml

```diff
@@ -0,0 +1,75 @@
+parameters:
+  jobNamePrefix: 'sourcebuild'    # Override for prefix for job names
+  managedOnly: false              # Specifies whether source is managed only. If so, one build platform is sufficient
+  portableOnly: false             # If true, builds one leg with portable flag enabled
+  prebuiltCheck: true             # Specifies whether to enforce prebuilt checking and fail if new prebuilts are introduced
+
+jobs:
+- template: /eng/common/templates/jobs/jobs.yml
+  parameters:
+    enablePublishUsingPipelines: true
+    enablePublishBuildArtifacts: true
+    enablePublishBuildAssets: true
+    artifacts:
+      publish:
+        artifacts: true
+        manifests: true
+    jobs:
+      # Only run fedora32 when either managedOnly or portableOnly is true.  Otherwise, run all legs.
+    - job: ${{ parameters.jobNamePrefix }}_fedora32
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-32-20200512010618-163ed2a'
+      workspace:
+        clean: all
+      steps:
+      - template: ../steps/run-build-from-source.yml
+        parameters:
+          portableOnly: ${{ parameters.portableOnly }}
+
+    - job: ${{ parameters.jobNamePrefix }}_centos71
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-3e800f1-20190501005343'
+      workspace:
+        clean: all
+      steps:
+      - template: ../steps/run-build-from-source.yml
+      condition: and( eq(${{ parameters.portableOnly }}, false), eq( ${{ parameters.managedOnly }}, false))
+
+    - job: ${{ parameters.jobNamePrefix }}_centos8
+      container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-8-daa5116-20200325130212'
+      workspace:
+        clean: all
+      steps:
+      - template: ../steps/run-build-from-source.yml
+      condition: and( eq(${{ parameters.portableOnly }}, false), eq( ${{ parameters.managedOnly }}, false))
```

Comment:
I don't like the duplication here... hard to spot differences. I'd look for something like:

```yml
parameters:
  standardPlatforms:
  - name: 'centos71'
    container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-3e800f1-20190501005343'
    # Indicates this is the standard platform: the one that should always build portable.
    standardPlatform: true
  nonPortablePlatforms:
  - name: 'debian9'
    container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:debian-stretch-d61254f-20190807161114'
  - name: 'fedora30'
    container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-30-38e0f29-20191126135223'

...

- template: ../jobs/build-from-source-jobs.yml
  parameters:
    platforms:
    - ${{ each platform in standardPlatforms }}:
      - ${{ platform }}
    - ${{ if ne(parameters.portableOnly, true) }}:
      - ${{ each platform in nonPortablePlatforms }}:
        - ${{ platform }}
```

Reply:
Eliminated duplication